pkg/routeagent/controllers/route/route.go

```diff
@@ -243,10 +476,19 @@ func (r *Controller) enqueueEndpoint(obj interface{}) {
 		utilruntime.HandleError(err)
 		return
 	}
-	klog.V(4).Infof("Enqueueing endpoint for route controller %v", obj)
 	r.endpointWorkqueue.AddRateLimited(key)
 }
 
+func (r *Controller) enqueuePod(obj interface{}) {
+	klog.V(6).Infof("Enqueueing pod for route controller %v", obj)
+	pod := obj.(*k8sv1.Pod)
+
+	// Add the POD event to the workqueue only if the sm-route-agent podIP does not exist in the local cache.
+	if !r.remoteVTEPs.Contains(pod.Status.HostIP) {
```

Comment:
I'd suggest doing this check in `processNextPod` for safety.

Reply:
yeah there can be hundreds of pods so shortcutting should be fine. However I'm wondering if we should only do this for an update...

Reply:
IMHO I don't see any major advantage with that. I feel its currently good and we can revisit this code if required.

Reply:
Current code looks ok for me too in this sense.